script.module.libmediathek4/lib/libmediathek4.py

```diff
@@ -0,0 +1,311 @@
+				liz.setCast(metadata['actors'])
+					
+			ok=True
+
+			if item['type'] in ['audio','songs']:
+				liz.setInfo( type="music", infoLabels=ilabels)
+			else:
+				liz.setInfo( type="Video", infoLabels=ilabels)
+				
+			if item.get('type',None) in ['video', 'live', 'date', 'clip', 'episode', 'audio', 'sport', 'sports', 'movie', 'song']:
+				liz.setProperty('IsPlayable', 'true')
+				lists.append([u,liz,False])
+			else:
+				lists.append([u,liz,True])
+
+		if 'content' in d:
+			xbmcplugin.setContent(handle=int(sys.argv[1]), content=d['content'] )
+		else:
+			xbmcplugin.setContent(handle=int(sys.argv[1]), content="files" )
+		
+		xbmcplugin.addDirectoryItems(int(sys.argv[1]), lists)
+
+	def play(self,d,external=False):
+		if not 'media' in d or len(d['media']) == 0:#TODO: add error msg
+			listitem = xbmcgui.ListItem(path='')
+			pluginhandle = int(sys.argv[1])
+			xbmcplugin.setResolvedUrl(pluginhandle, False, listitem)
+			return
+
+		listitem,url = self._chooseBitrate(d['media'])	
+				
+		if 'subtitle' in d:
+			subs = []
+			for subtitle in d['subtitle']:
+				if subtitle['type'] == 'srt':
+					subs.append(subtitle['url'])
+				elif subtitle['type'] == 'ttml':
+					import libmediathek4ttml2srt
+					subFile = libmediathek4ttml2srt.ttml2Srt(subtitle['url'])
+					subs.append(subFile)
+				elif subtitle['type'] == 'webvtt':
+					import libmediathek4webvtt2srt 
+					subFile = libmediathek4webvtt2srt.webvtt2Srt(subtitle['url'])
+					subs.append(subFile)
+				else:
+					log('Subtitle format not supported: ' + subtitle['type'])
+			listitem.setSubtitles(subs)
+		
+		if 'metadata' in d:
+			ilabels = {}
+			if 'plot' in d['metadata']:
+				ilabels['Plot'] = d['metadata']['plot']
+			if 'name' in d['metadata']:
+				ilabels['Title'] = d['metadata']['name']
+			listitem.setInfo( type="Video", infoLabels=ilabels)
+			
+			art = {}
+			if 'thumb' in d['metadata']:
+				art['thumb'] = d['metadata']['thumb']
+			listitem.setArt(art)
+			
+		if 'header' in d['media']:
+			listitem.setProperty('inputstream.adaptive.stream_headers',d['media']['header'])
+		
+		if external:
+			xbmc.Player().play(url, listitem)
+		else:
+			pluginhandle = int(sys.argv[1])
+			xbmcplugin.setResolvedUrl(pluginhandle, True, listitem)
+			
+	def _chooseBitrate(self,l):
+		bitrate = 0
+		url = False
+		streamType = False
+		for item in l:
+			if item.get('stream','').lower() == 'hls':#prefer hls
+				url = item['url']
+				streamType = 'HLS'
+				break
+			if item.get('stream','').lower() == 'dash':
+				url = item['url']
+				streamType = 'DASH'
+			if item.get('stream','').lower() == 'mp4' and item.get('bitrate',0) >= bitrate:
+				bitrate = item.get('bitrate',0)
+				url = item['url']
+				streamType = 'MP4'
+			if item.get('stream','').lower() == 'audio':
+				url = item['url']
+				streamType = 'AUDIO'
+		listitem = xbmcgui.ListItem(path=url)
+		if streamType == 'DASH':
+			listitem.setProperty('inputstreamaddon', 'inputstream.adaptive')

```

Comment:
For Matrix this should now be: `listitem.setProperty('inputstream', 'inputstream.adaptive')`

https://forum.kodi.tv/showthread.php?tid=353560

Reply:
@sabres this one still missing